src/py/flwr/common/secure_aggregation/secaggplus_constants.py

```diff
@@ -14,33 +14,55 @@
 # ==============================================================================
 """Constants for the SecAgg/SecAgg+ protocol."""
 
+
+from __future__ import annotations
+
 RECORD_KEY_STATE = "secaggplus_state"
 RECORD_KEY_CONFIGS = "secaggplus_configs"
 
-# Names of stages
-STAGE_SETUP = "setup"
-STAGE_SHARE_KEYS = "share_keys"
-STAGE_COLLECT_MASKED_INPUT = "collect_masked_input"
-STAGE_UNMASK = "unmask"
-STAGES = (STAGE_SETUP, STAGE_SHARE_KEYS, STAGE_COLLECT_MASKED_INPUT, STAGE_UNMASK)
-
-# All valid keys in received/replied `named_values` dictionaries
-KEY_STAGE = "stage"
-KEY_SAMPLE_NUMBER = "sample_num"
-KEY_SECURE_ID = "secure_id"
-KEY_SHARE_NUMBER = "share_num"
-KEY_THRESHOLD = "threshold"
-KEY_CLIPPING_RANGE = "clipping_range"
-KEY_TARGET_RANGE = "target_range"
-KEY_MOD_RANGE = "mod_range"
-KEY_PUBLIC_KEY_1 = "pk1"
-KEY_PUBLIC_KEY_2 = "pk2"
-KEY_DESTINATION_LIST = "dsts"
-KEY_CIPHERTEXT_LIST = "ctxts"
-KEY_SOURCE_LIST = "srcs"
-KEY_PARAMETERS = "params"
-KEY_MASKED_PARAMETERS = "masked_params"
-KEY_ACTIVE_SECURE_ID_LIST = "active_sids"
-KEY_DEAD_SECURE_ID_LIST = "dead_sids"
-KEY_SECURE_ID_LIST = "sids"
-KEY_SHARE_LIST = "shares"
+
+class Stage:
+    """Stages for the SecAgg+ protocol."""
+
+    SETUP = "setup"
+    SHARE_KEYS = "share_keys"
+    COLLECT_MASKED_INPUT = "collect_masked_input"
+    UNMASK = "unmask"
+    _stages = (SETUP, SHARE_KEYS, COLLECT_MASKED_INPUT, UNMASK)
+
+    @classmethod
+    def all(cls) -> tuple[str, str, str, str]:
```

Comment:
If we use a `tuple` here, we'll have to change the type if new elements are introduces. We could also use a `Set[str]`

Reply:
I think we prolly don't need to change it. The stage follows the SecAgg+ protocol. It's not defined by us.